### ISSUE TITLE
Oil refining, hear me out

### ISSUE THREAD

**Issue by alexHunter854** [Anuken/Mindustry-Suggestions#2783]
### Describe the content or mechanics you are proposing.

Oil refinery:
3×3 late game block which takes in power and crude oi and turns it into a small amount of gas (and maybe some asphalt, which could be an interesting item if you could find a use for it, maybe making shallow water into road the walking units can cross. Naphthacould also be a byproduct that couldbe used to replace oil in plastanium production).

Gas generator: 
3×3 late game block that turns gas into power, must also have water as coolent, although can be replaced with cryo.

Gas: can be used as a booster for units, turrets and drills, maybe even a requirement for some drills. Could also be used as a requirement for heigh tier units instead of optional booster. Could maybe be used as rocket fuel for the nuclear missile you're working on. 

### Describe how you think this content will improve the game. If you're proposing new content, mention how it may add more gameplay options or how it will fill a new niche.

I think that if there's oil in a game the next natural evolution is refining that oil, it will open up new possibilities for development. It would be more efficient than oil -> coal -> power.

### Before making this issue, check the boxes below to confirm that you have acknowledged them.

- [X] I have checked the [Trello](https://trello.com/b/aE2tcUwF/mindustry-trello) to make sure my suggestion isn't planned or implemented in a development version.
- [X] I am familiar with all the content already in the game or have glanced at the wiki to make sure my suggestion doesn't exist in the game yet.
- [X] I have read [README.md](https://github.com/Anuken/Mindustry-Suggestions/blob/master/README.md) to make sure my idea is not listed under the "A few things you shouldn't suggest" category.

**Comment by MrDuck557** [comment]
whee
extra complexity

**Comment by oinkymoo** [comment]
i feel like this would probably be like the next tier of tech if it got implemented

**Comment by GitHubUser-B** [comment]
> whee
> extra complexity

What is the problem!? Can't the game be more challenging and have more mechanics?

**Comment by Kyllingene** [comment]
Good games don't keep adding mechanics (at least, most of them don't); good games get as much gameplay as they can out of as few mechanics as possible. So no, added complexity is *not* a good thing; this adds almost no gameplay while adding a lot of complexity.

**Comment by GitHubUser-B** [comment]
> Good games don't keep adding mechanics (at least, the majority don't); good games get as much gameplay as they can out of as few mechanics as possible. So no, added complexity is _not_ a good thing; this adds almost no gameplay while adding a lot of complexity. Thus, this would be a fantastic mod, but a very bad built-in gameplay feature.

Factorio has more mechanics than this game, but it's still very fun to play! (The community is kinder too)

**Comment by OnyxFinger** [comment]
> > Good games don't keep adding mechanics (at least, the majority don't); good games get as much gameplay as they can out of as few mechanics as possible. So no, added complexity is _not_ a good thing; this adds almost no gameplay while adding a lot of complexity. Thus, this would be a fantastic mod, but a very bad built-in gameplay feature.
> 
> Factorio has more mechanics than this game, but it's still very fun to play! (The community is kinder too)

but it is way more complicated so the current players of mindustry's player base wouldn't like it (because most players play this game for it's simplicity)
also because a game has complicated mechanics and is also good doesn't mean it's a good idea to make every game complex

**Comment by GitHubUser-B** [comment]
> > > Good games don't keep adding mechanics (at least, the majority don't); good games get as much gameplay as they can out of as few mechanics as possible. So no, added complexity is _not_ a good thing; this adds almost no gameplay while adding a lot of complexity. Thus, this would be a fantastic mod, but a very bad built-in gameplay feature.
> > 
> > 
> > Factorio has more mechanics than this game, but it's still very fun to play! (The community is kinder too)
> 
> (because most players play this game for it's simplicity)

Seriously? I did not know that!

**Comment by Kyllingene** [comment]
"Hey, this somewhat similar game has X trait that this game doesn't have 7 releases in, so we should add it too, regardless of how it works with the rest of the game!"

You do realize how hard it is for most people to get into Factorio, right? Why do you think we need to ostracize a large part of the community by replacing a perfectly fine system with an overcomplicated one? 

Again, this is not a bad idea in and of itself, but most of the fanbase is not looking for Factorio v2; if they were, they would be playing Factorio.

**Comment by GitHubUser-B** [comment]
> "Hey, this somewhat similar game has X trait that this game doesn't have 7 releases in, so we should add it too, regardless of how it works with the rest of the game!"
> 
> You do realize how hard it is for most people to get into Factorio, right? Why do you think we need to ostracize a large part of the community by replacing a perfectly fine system with an overcomplicated one?
> 
> Again, this is not a bad idea in and of itself, but most of the fanbase is not looking for Factorio v2; if they were, they would be playing Factorio.
> 
> Also, the products couldn't be implemented; factories can't (and won't) accept an alternative ingredient, we can't build over liquid like that, we aren't implementing a whole new transport system for gases, nor can we boost units, we don't need another block booster, we don't need another building requirement for units, and I highly doubt the rocket will need refueling, and it could also use blast compound as fuel anyway.
> 
> ## TL;DR: the system is unnecessary and the products even more so

I also thought that it has a relation with Factorio, they are both cool!

**Comment by OnyxFinger** [comment]
> > > > Good games don't keep adding mechanics (at least, the majority don't); good games get as much gameplay as they can out of as few mechanics as possible. So no, added complexity is _not_ a good thing; this adds almost no gameplay while adding a lot of complexity. Thus, this would be a fantastic mod, but a very bad built-in gameplay feature.
> > > 
> > > 
> > > Factorio has more mechanics than this game, but it's still very fun to play! (The community is kinder too)
> > 
> > 
> > (because most players play this game for it's simplicity)
> 
> Seriously? I did not know that!

then don't bring up complexity as an argument :confused:

**Comment by GitHubUser-B** [comment]
> > > > > Good games don't keep adding mechanics (at least, the majority don't); good games get as much gameplay as they can out of as few mechanics as possible. So no, added complexity is _not_ a good thing; this adds almost no gameplay while adding a lot of complexity. Thus, this would be a fantastic mod, but a very bad built-in gameplay feature.
> > > > 
> > > > 
> > > > Factorio has more mechanics than this game, but it's still very fun to play! (The community is kinder too)
> > > 
> > > 
> > > (because most players play this game for it's simplicity)
> > 
> > 
> > Seriously? I did not know that!
> 
> then don't bring up complexity as an argument 😕

What do you mean argument? I just wanted to know why the duck complained about it 😕

**Comment by Kyllingene** [comment]
Is there anything more to talk about? Either we showed conclusively that this is unnecessary, or you need to show us how we didn't.

**Comment by GitHubUser-B** [comment]
> Is there anything more to talk about? Either we showed conclusively that this is unnecessary and you need to close this, or we didn't and you need to show us how.

![Jymahvz](https://user-images.githubusercontent.com/77138172/129926141-04e1913e-cbaf-4037-8f6c-5f2b0bd11788.png)
